### PR TITLE
feat: Implement runtine check for enterprise features

### PR DESCRIPTION
**Pull request by krynble**

### CODE REVIEW

File: packages/cli/src/UserManagement/PermissionChecker.ts [hunk 1]
```diff
@@ -26,23 +29,30 @@ export class PermissionChecker {
...
-		});
+		const credentialsWhereCondition: FindManyOptions<SharedCredentials> & { where: ObjectLiteral } =
+			{
+				where: { user: In(workflowUserIds) },
+			};
+
+		if (!config.getEnv('enterprise.features.sharing')) {
+			// If credential sharing is not enabled, get only credentials owned by this user
+			const credentialOwnerRole = await getRole('credential', 'owner');
+			credentialsWhereCondition.where.role = credentialOwnerRole;
```
Comment (ivov) [inline packages/cli/src/UserManagement/PermissionChecker.ts:50]:
```suggestion
			credentialsWhereCondition.where.role = await getRole('credential', 'owner');
```

File: packages/cli/src/UserManagement/UserManagementHelper.ts [hunk 2]
```diff
@@ -78,6 +78,16 @@ export async function getInstanceOwner(): Promise<User> {
...
 }
 
+export async function getRole(scope: RoleScopes, name: RoleNames): Promise<Role> {
+	const role = await Db.collections.Role.findOneOrFail({
+		where: {
+			name,
+			scope,
+		},
+	});
+	return role;
```
Comment (ivov) [inline packages/cli/src/UserManagement/UserManagementHelper.ts:88]:
```suggestion
	return await Db.collections.Role.findOneOrFail({
		where: {
			name,
			scope,
		},
	});
```

File: packages/cli/src/UserManagement/UserManagementHelper.ts [hunk 3]
```diff
@@ -78,6 +78,16 @@ export async function getInstanceOwner(): Promise<User> {
 	return owner;
 }
 
+export async function getRole(scope: RoleScopes, name: RoleNames): Promise<Role> {
```
Comment (ivov) [inline packages/cli/src/UserManagement/UserManagementHelper.ts:81]: You can also use lookup types to avoid ex/importing: `Role['name']`, `Role['scope']`.

**Review by ivov — DISMISSED** [review]
All tests passing on all three DBs. Only minor comments.

**Review by ivov — APPROVED** [review]
🚀

**Review by ivov — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.204.0`

**Pull request merged**